src/ssh-generator/ssh-proxy.c

```diff
@@ -91,6 +91,129 @@ static int process_unix(const char *path) {
         return 0;
 }
 
+static int skip_ok_port_res(int fd, const char *path, const char *port) {
+        struct timeval oldtv;
+        socklen_t oldtv_size = sizeof(oldtv);
+        if (getsockopt(fd, SOL_SOCKET, SO_RCVTIMEO, &oldtv, &oldtv_size) < 0)
+                return log_error_errno(errno, "Failed to get socket receive timeout for %s: %m", path);
+        if (oldtv_size != sizeof(oldtv))
+                return log_error_errno(SYNTHETIC_ERRNO(EIO), "Unexpected size of socket receive timeout for %s", path);
+        struct timeval newtv = {
+                .tv_sec = 5,
+                .tv_usec = 0,
```

Comment:
nitpick, we typically don't bother initializing zeroes when using compound initialization, as the compiler does that anyway for unspecified fields.

btw, it might make sense to write this as:

```c
if (setsockopt(fd, SOL_SOCKET, SO_RECVTIMEOU, TIMEVAL_STORE(LISTEN_TIMEOUT_USEC), sizeof(struct timeval)) < 0) …
```

it's how we do that for all other uses of SO_RCVTIMEO